src/asl/synth.ts

```diff
@@ -1974,6 +1974,56 @@ export class ASL {
               : ASLGraph.intrinsicStringToJson(output)
           );
         });
+      } else if (isSplit(expr)) {
+        const [splitter, limit] = expr.args;
+
+        if (limit) {
+          throw new SynthError(
+            ErrorCodes.Unsupported_Feature,
+            "Step Functions does not support the limit parameter on String.split"
+          );
+        } else if (!splitter) {
+          throw new SynthError(
+            ErrorCodes.Invalid_Input,
+            "Step Functions String Split splitter argument is required"
+          );
+        }
```

Comment:
Is it really called splitter and not delimiter? Haha

Reply:
yes. Typescript/ECMA calls it separator or splitter.